oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/OAuth2AuthorizationServerConfiguration.java

```diff
@@ -88,9 +89,10 @@ public static JwtDecoder jwtDecoder(JWKSource<SecurityContext> jwkSource) {
 	}
 
 	@Bean
-	RegisterMissingBeanPostProcessor registerMissingBeanPostProcessor() {
+	RegisterMissingBeanPostProcessor registerMissingBeanPostProcessor(@Value("${server.port:9000}") Integer port) {
 		RegisterMissingBeanPostProcessor postProcessor = new RegisterMissingBeanPostProcessor();
-		postProcessor.addBeanDefinition(ProviderSettings.class, () -> ProviderSettings.builder().build());
+		postProcessor.addBeanDefinition(ProviderSettings.class,
```

Comment:
There is no need to explicitly default the `scheme`, `host` or `port` since this is handled dynamically via `ProviderContextFilter` - see gh-479

Reply:
Thanks, I ignored this important yet.